Sources/TuistSupport/Utils/RecentPathsStore.swift

```diff
@@ -0,0 +1,78 @@
+import FileSystem
+import Foundation
+import Mockable
+import Path
+import ServiceContextModule
+
+private enum RecentPathsContextKey: ServiceContextKey {
+    typealias Value = RecentPathsStoring
+}
+
+extension ServiceContext {
+    public var recentPaths: RecentPathsStoring? {
+        get {
+            self[RecentPathsContextKey.self]
+        } set {
+            self[RecentPathsContextKey.self] = newValue
+        }
+    }
+}
+
+/// This is a utility to record the paths the user interacts with.
+/// This is recorded either from the working directory or the --path argument when invoking a command.
+/// The information can then be used by tools like MCP servers to allow users to interact with their most recent
+/// projects.
+@Mockable
+public protocol RecentPathsStoring {
+    /// Records that the user has interacted with a given path.
+    /// - Parameters:
+    ///   - path: The path the user has interacted with.
+    ///   - date: The date of interaction
+    func remember(path: AbsolutePath, date: Date) async throws
+
+    /// Returns the list of paths the user has interacted with along with the last date of interaction.
+    /// - Returns: A dictionary where the keys are the paths, and the values are the last time the user interacted with those
+    /// paths.
+    func read() async throws -> [AbsolutePath: Date]
```

Comment:
What do you think of making the value of the dictionary a struct? I can see us putting additional metadata such as the name of the project. If we make it a struct now, it will be easier to keep it backwards compatible. Since we're writing to a file, we need to ensure we can read files written with older schemas.

Reply:
Sounds good! I adjusted it to have its own type.